CMakeLists.txt

```diff
@@ -17,7 +17,7 @@ if (ENABLE_STACK_DEBUG)
 endif()
 
 if(CMAKE_COMPILER_IS_GNUCXX OR "${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
-    set(CMAKE_CXX_FLAGS "-Wall -Wextra -Wshadow -Werror ${CMAKE_CXX_FLAGS}")
+    set(CMAKE_CXX_FLAGS "-Wall -Wextra -Werror ${CMAKE_CXX_FLAGS}")
```

Comment:
This will disable all types of shadowing warnings, not just the overzealous one that GCC 4.8 had...  I would prefer the parameters be renamed and this warning kept to make GCC happy.

Reply:
makes sense!
agree, but would it be fine with you if I rename all those shadowed input params (mostly named as `type` into `_type`) ?

Reply:
Yeah, that would be fine